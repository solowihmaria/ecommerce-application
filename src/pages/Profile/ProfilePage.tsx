import { UserProfile } from '../../components/blocks/UserProfile';
import styles from './ProfilePage.module.scss';

export const ProfilePage = () => {
    return (
        <div data-testid="profile-page-test-id" className={styles.profilePage}>
            <UserProfile />
        </div>
    );
};
