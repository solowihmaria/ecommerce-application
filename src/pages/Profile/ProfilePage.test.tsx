import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProfilePage } from './ProfilePage';
import { MemoryRouter } from 'react-router-dom';

describe('<ProfilePage/> component', () => {
    test('renders ProfilePage component', () => {
        render(
            <MemoryRouter>
                <ProfilePage />
            </MemoryRouter>
        );

        const profileMain = screen.getByRole('main');
        expect(profileMain).toBeInTheDocument();
        expect(profileMain).toMatchSnapshot();
    });
});
