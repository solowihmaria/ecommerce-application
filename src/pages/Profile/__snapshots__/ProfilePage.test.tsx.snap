// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePage/> component renders ProfilePage component 1`] = `
<main
  class="profileContent"
>
  <div
    class="profileContainer"
  >
    <div
      class="headerWrapper"
    >
      <svg
        class="profileIcon"
        fill="none"
        height="1em"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
        />
        <circle
          cx="12"
          cy="7"
          r="4"
        />
      </svg>
      <h1
        class="heading heading--h1 pageTitle"
      >
        My Profile
      </h1>
    </div>
    <div
      class="profileMain"
    >
      <div
        class="infoSection"
      >
        <div>
          Loading user data...
        </div>
      </div>
      <div
        class="passwordSection"
      >
        <div
          class="changePassword"
        >
          <h2
            class="heading heading--h2 title"
          >
            Change Password
          </h2>
          <form
            class="passwordForm"
          >
            <div
              class="field"
            >
              <label
                class="label"
                for="currentPassword"
              >
                Current Password
              </label>
              <div
                class="inputContainer"
              >
                <div
                  class="inputWrapper"
                >
                  <input
                    class="input password"
                    id="currentPassword"
                    name="currentPassword"
                    type="password"
                  />
                  <button
                    aria-label="Show password"
                    class="toggleButton"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="field"
            >
              <label
                class="label"
                for="newPassword"
              >
                New Password
              </label>
              <div
                class="inputContainer"
              >
                <div
                  class="inputWrapper"
                >
                  <input
                    class="input password"
                    id="newPassword"
                    name="newPassword"
                    type="password"
                  />
                  <button
                    aria-label="Show password"
                    class="toggleButton"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="field"
            >
              <label
                class="label"
                for="confirmPassword"
              >
                Confirm New Password
              </label>
              <div
                class="inputContainer"
              >
                <div
                  class="inputWrapper"
                >
                  <input
                    class="input password"
                    id="confirmPassword"
                    name="confirmPassword"
                    type="password"
                  />
                  <button
                    aria-label="Show password"
                    class="toggleButton"
                    type="button"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 576 512"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
            <div
              class="formActions"
            >
              <button
                class="button outline submitButton disabled"
                disabled=""
                type="submit"
              >
                Update Password
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      class="addressesSection"
    >
      <div>
        Loading addresses...
      </div>
    </div>
  </div>
</main>
`;
